### ISSUE TITLE
Datadog Agent stops reporting metrics and consumes 100% CPU on Mac OS

### ISSUE THREAD

**Issue by goodspark** [DataDog/datadog-agent#6539]
```
{
	"StandardOutPath" = "/var/log/datadog/launchd.log";
	"LimitLoadToSessionType" = "Aqua";
	"StandardErrorPath" = "/var/log/datadog/launchd.log";
	"Label" = "com.datadoghq.agent";
	"OnDemand" = true;
	"LastExitStatus" = 19968;
	"PID" = 35860;
	"Program" = "/opt/datadog-agent/bin/agent/agent";
	"ProgramArguments" = (
		"/opt/datadog-agent/bin/agent/agent";
		"run";
	);
};
```

**Describe what happened:**
The agent stopped reporting metrics for the host (eg. CPU/memory) all the while consuming 100% CPU.

**Describe what you expected:**
The agent should continuously report metrics. The agent shouldn't be consuming 10% CPU.

**Steps to reproduce the issue:**
Unsure. I simply left the agent running on this machine over night.

**Additional environment details (Operating System, Cloud provider, etc):**
Mac Mini
Mac OS 10.15.7, build 19H2
Intel(R) Core(TM) i7-8700B CPU @ 3.20GHz
16GB RAM
Datadog Agent 7.23.0 - Commit: 3cc240ddd - Serialization version: v4.44.0 - Go version: go1.14.7

Here are no logs in /var/log/datadog/launchd.log, nothing relevant in /var/log/system.log and no error/warning logs in /var/log/datadog/agent.log.

**Comment by mx-psi** [comment]
As stated in issue #6538, we believe this issue to be related to golang/go#39079 and we are working on a fix.

**Comment by mx-psi** [comment]
As stated in issue #6538, the recently released Agent version 7.25.0 should fix this issue and issue #6538 since we upgraded Go (see #6942). Feel free to comment or reopen if you find any of the issues again after upgrading.
